imageutils.py

```diff
@@ -246,17 +251,21 @@ def split_clustered_writes(self, Y_splits, Z_splits, X_splits, out_dir, mem, fil
                 x_e = split_start[2] + x_size
 
                 split_data = data[split_start[0] : y_e, split_start[1] : z_e, split_start[2] : x_e]
-                im = nib.Nifti1Image(split_data, self.affine)
+                # we cannot save .nii image to hdfs this way.
```

Comment:
You're loading an image into nibabel here from a reconstructed image stored locally, so it would work. Would also work if you extracted the image data from HDFS, as it's using data in-memory to create a nibabel object.